runtime/bcutil/ComparingCursor.cpp

```diff
@@ -109,7 +110,17 @@ ComparingCursor::writeU32(U_32 u32Value, DataType dataType)
 	ComparingCursorHelper * countingcursor = getCountingCursor(dataType);
 	if ( shouldCheckForEquality(dataType, u32Value) ) {
 		U_32 * tmpu32 = (U_32 *)(countingcursor->getBaseAddress() + countingcursor->getCount());
-		if ( !isRangeValid(sizeof(U_32), dataType) || (u32Value != *tmpu32) ) {
+		/* if the class is a lambda class and the data type is class file size,
+		 * we are not actually checking if it is equal but checking if the
+		 * difference is less than sizeof(U_32) to match lambda classes that have
+		 * different sized class names (e.g., Test1$$Lambda$9/0000000000000000 and Test1$$Lambda$10/0000000000000000)
+		 */
+		if(CLASS_FILE_SIZE == dataType && isLambda()){
```

Comment:
Coding standard is here https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md. There should be a space between "if" and "(", ")" and "{". Please fix this in all the lines you changed.